src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -343,27 +343,41 @@ protected internal override void Scan(ReadOnlySpan<char> text)
             // to call TryFindNextStartingPosition again, as well as the stopping position for the loop.  We generally
             // bump by 1 and stop at textend, but if we're examining right-to-left, we instead bump
             // by -1 and stop at textbeg.
-            int bump = 1, stoppos = text.Length;
-            if (runregex.RightToLeft)
+            if (!runregex.RightToLeft)
```

Comment:
We should probably also update the comment above. 

Also, super small NIT, but in order to make it a bit more readable, can we keep the structure like:
```c#
 if (runregex.RightToLeft)
 {
     ...TryFindNextStartingPisitionRightToLeft()...
  }
   else
   {
      ..TryFindNextStartingPositionLeftToRight()..
```
Super dumb thing, but for some reason having the condition negated and for that to match the overload called in the else statement looks weird :P

Reply:
Addressed, thanks